src/infra/bus/distributed.js

```diff
@@ -0,0 +1,378 @@
+
+    this.#logger.info(
+      { callId, result, subKey },
+      `Replied - ${service}/${method}`,
+    );
+  };
+
+  /** @type ICommand['call'] */
+  #localCall = async ({ service: serviceName, method }, payload) => {
+    const service = this.#services.get(serviceName);
+    if (!service) {
+      return [{ expected: false, message: 'Service not found' }, null];
+    }
+
+    const handler = service[method];
+    if (!handler) {
+      return [{ expected: false, message: 'Method not found' }, null];
+    }
+
+    const result = await handler(payload);
+    return result;
+  };
+
+  #listenRemoteEvents = async () => {
+    const eventNames = this.#eventHandlers.keys();
+    const streams = [];
+    for (const event of eventNames) {
+      const key = `${event}:event`;
+      streams.push({ key, id: '>' });
+    }
+    if (!streams.length) return;
+    const redis = this.#redis.duplicate();
+    await redis.connect();
+    do {
+      await this.#processGroupStreams(
+        redis,
+        this.#eventsGroupName,
+        streams,
+        this.#handleRemoteEvent,
+      );
+    } while (!this.#terminating);
+    await redis.quit();
+  };
+
+  /** @type {(streamName: string, data: CallData) => Promise<void>} */
+  #handleRemoteEvent = async (streamName, { payload: strPayload }) => {
+    const eventName = streamName.slice(0, streamName.lastIndexOf(':'));
+    const payload = JSON.parse(strPayload);
+
+    this.#logger.info({ streamName, payload }, `Received event - ${eventName}`);
+
+    const handler = this.#eventHandlers.get(eventName);
+    if (!handler) {
+      this.#logger.warn(
+        { streamName, payload, eventName },
+        'Missing event handler',
+      );
+      return;
+    }
+
+    await handler(payload);
+
+    this.#logger.info(`Processed event - ${eventName}`);
+  };
+
+  /** @type {(
+   * redis: Redis,
+   * group: string,
+   * streams: { key: string; id: string; }[],
+   * processor: (streamName: string, data: CallData) => Promise<any>,
+   * ) => Promise<void>} */
+  #processGroupStreams = async (redis, group, streams, processor) => {
+    const { serverId: consumer, readInterval } = this.#options;
+    const events = await redis.xReadGroup(group, consumer, streams, {
+      COUNT: 1,
+      BLOCK: readInterval,
+      NOACK: true,
+    });
+    if (!events?.length) return;
+    const { name: streamName, messages } = events[0];
+    const { message } = messages[0];
+    const payload = /** @type CallData */ (message);
+    const process = processor(streamName, payload);
+    const processId = randomUUID();
+    this.#processing.set(processId, process);
+    process.then(() => this.#processing.delete(processId));
+  };
+
+  /** @type IBus['withMeta'] */
+  withMeta = (meta) => {
+    const allowedMethods = ['call', 'publish'];
+    return new Proxy(this, {
+      get(target, prop) {
+        if (typeof prop === 'string' && allowedMethods.includes(prop)) {
+          return undefined;
+        }
```

Comment:
this if statement must be reversed. call and publish are both type `string` and in `allowedMethods` but undefined returns!!!
```js
if (typeof prop !== 'string' || !allowedMethods.includes(prop)) ...
```